pkg/service/oidc4vp/attachments.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+//go:generate mockgen -destination attachments_mocks_test.go -self_package mocks -package oidc4vp_test -source=attachments.go -mock_names httpClient=MockHttpClient
+
+package oidc4vp
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+
+	"github.com/samber/lo"
+	"github.com/trustbloc/vc-go/util/maphelpers"
+	"github.com/trustbloc/vc-go/verifiable"
+)
+
+const (
+	AttachmentTypeRemote   = "RemoteAttachment"
+	AttachmentTypeEmbedded = "EmbeddedAttachment"
+	AttachmentDataField    = "uri"
```

Comment:
how about hash & hash-alg?

Reply:
yep, i will add checks for hash in next pr, thanks